lib/api/CDataFrameAnalysisInstrumentation.cc

```diff
@@ -267,9 +267,6 @@ void CDataFrameTrainBoostedTreeInstrumentation::writeAnalysisStats(std::int64_t
         case E_Classification:
             writer->Key(CLASSIFICATION_STATS_TAG);
             break;
-        default:
-            LOG_ERROR(<< "Supervised learning type unknown or not set.");
-            return;
```

Comment:
++ to removing this.

With `clang++` we build with the `-Werror-switch` option so if another enum value was added to the enum definition but not added to the switch then the file wouldn't compile on macOS, so we would catch it in the PR build at the latest.